pyproject.toml

```diff
@@ -127,12 +126,12 @@ find = {}
 #######################################################################
 [tool.ruff]
 # This is the exact version of Ruff we use.
-required-version = "0.5.1"
+required-version = "0.9.7"
 
 # Assume Python 3.9
-target-version = "py39"
+target-version = "py313"
 
-# Setting line-length to 120 (though blacks default is 88)
+# Setting line-length to 120
 line-length = 120
```

Comment:
you sure you don't want to take this opportunity to go to the ruff and black default of 88? 88 is great! https://docs.astral.sh/ruff/settings/#line-length https://stackoverflow.com/a/578318/1279313

otherwise this looks great.

Reply:
I tried. Even going down to 100 meant I had over 400 files that required manual changes. (Things like docstrings couldn't be automatically reformatted.)

And I do not have time for manually touching >1,000 lines to make them shorter.

(Also, as a personal preference, I might go 100, but 88 feels a bit too narrow when you have a lot of math on the same line.)